Sources/Sentry/SentryScope.m

```diff
@@ -319,6 +326,33 @@ - (void)addAttachment:(SentryAttachment *)attachment
     }
 }
 
+- (void)setTransaction:(id<SentrySpan>)transaction
+{
+    self.currentTransaction = transaction;
+    [self notifyListeners];
+}
+
+- (void)setTransactionName:(NSString *)transactionName
+{
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]]) {
```

Comment:
there's a race condition here.
By the time you're setting the name to the current transaction below the instance could already have changed.

You'd need to synchrnize access to this thing across all of the scope. Which is anyway flawed since anyway the reference can be accessed, and then `transaction.name` can be set outside of this.

I believe this is a whole in the API also because any code setting `scope.transaction` that existed before due to crash reporting will start messing around with a transaction that was created/set in the scope. 

No suggestions on how to fix this, just think it's broken right now though.

Reply:
setTransactionName was removed.